examples/correlated_ou/analysis.jl

```diff
@@ -10,91 +10,115 @@ using Statistics
 using TOML
 
 using CliMAgen
+
 package_dir = pkgdir(CliMAgen)
-include(joinpath(package_dir,"examples/utils_data.jl"))
-include(joinpath(package_dir,"examples/utils_analysis.jl"))
+include(joinpath(package_dir, "examples/utils_data.jl"))
+include(joinpath(package_dir, "examples/utils_analysis.jl"))
 
 function run_analysis(params; FT=Float32, logger=nothing)
     # unpack params
     savedir = params.experiment.savedir
     rngseed = params.experiment.rngseed
     nogpu = params.experiment.nogpu
-    
+
     resolution = params.data.resolution
+    ntime = params.data.ntime
     fraction = params.data.fraction
     standard_scaling = params.data.standard_scaling
     preprocess_params_file = joinpath(savedir, "preprocessing_standard_scaling_$standard_scaling.jld2")
-    
-    inchannels = params.model.noised_channels
+
+    inchannels = params.model.inchannels
+
     nsamples = params.sampling.nsamples
-    nimages = params.sampling.nimages
+    ngifs = params.sampling.ngifs
     nsteps = params.sampling.nsteps
     sampler = params.sampling.sampler
-    tilesize_sampling = params.sampling.tilesize
 
     # set up rng
     rngseed > 0 && Random.seed!(rngseed)
 
     # set up device
     if !nogpu && CUDA.has_cuda()
-        device = Flux.gpu
+        dev = Flux.gpu
         @info "Sampling on GPU"
     else
-        device = Flux.cpu
+        dev = Flux.cpu
         @info "Sampling on CPU"
     end
 
     # set up dataset
-    dl, _ = get_data_correlated_ou2d(
+    dl, _ = get_data_correlated_ou2d_timeseries(
         nsamples;
         f = fraction,
         resolution=resolution,
+        ntime=ntime,
         FT=Float32,
         standard_scaling = standard_scaling,
         read = true,
         save = false,
         preprocess_params_file,
         rng=Random.GLOBAL_RNG
     )
-    xtrain = first(dl)
+    xtrain, dl = Iterators.peel(dl)
+    nspatial=3
 
     # set up model
     checkpoint_path = joinpath(savedir, "checkpoint.bson")
     BSON.@load checkpoint_path model model_smooth opt opt_smooth
-    model = device(model)
-    
+    model = dev(model)
+
     # sample from the trained model
-    time_steps, Δt, init_x = setup_sampler(
-        model,
-        device,
-        tilesize_sampling,
-        inchannels;
-        num_images=nsamples,
-        num_steps=nsteps,
-    )
-    if sampler == "euler"
-        samples = Euler_Maruyama_sampler(model, init_x, time_steps, Δt)
-    elseif sampler == "pc"
-        samples = predictor_corrector_sampler(model, init_x, time_steps, Δt)
-    end
+    t = ones(FT, nsamples) |> dev
```

Comment:
do we need a `setup_sampler` for nspatial = 3? or we can probably adjust the one we accept `size(x)` as an argument? https://github.com/CliMA/CliMAgen.jl/blob/6b8f1de1b53593dca5c5ced6970ce729f1a0c702/src/sampling.jl#L182

Reply:
Could. do :)